### PR TITLE
Crude Results Grid

### PR DESCRIPTION
**Pull request by stdavis**
The UI is super-ugly. But the data seems to be loading correctly. No fancy paging. It seems to be fine. I plan on addressing the look and feel in my next PR.

### CODE REVIEW

**Comment by steveoh** [comment]
Is this caused by the preview environment variables?
image

What's a good filter to use to view the table?

File: package.json [hunk 1]
```diff
@@ -10,19 +10,18 @@
...
-    "dev": "npm start",
-    "dev:vite": "wait-on http://127.0.0.1:4000 && vite",
     "dev:firebase-emulators": "firebase emulators:start --import emulator_data --export-on-exit --only functions,auth,ui,firestore",
+    "dev:vite": "wait-on http://127.0.0.1:4000 && vite",
     "format": "prettier . --write",
     "lint": "eslint . --report-unused-disable-directives --max-warnings 0",
-    "check": "tsc -b",
     "preview": "vite preview",
-    "start": "run-p dev:*",
+    "start": "concurrently 'npm run dev:vite' 'npm run dev:firebase-emulators'",
```
Comment (steveoh) [inline package.json:21]: **question**: oh fun, what made you switch?

File: firebase.json [hunk 2]
```diff
@@ -66,14 +66,22 @@
...
+      {
+        "source": "**",
+        "destination": "/index.html"
+      },
+      {
+        "source": "/maps",
+        "function": {
+          "functionId": "maps",
+          "region": "us-central1",
+          "pinTag": true
```
Comment (steveoh) [inline firebase.json:80]: nice!

File: src/components/utils.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { composeRenderProps } from 'react-aria-components';
```
Comment (steveoh) [inline src/components/utils.ts:1]: **question**: I assume this is temporary while working on the table?

File: vite.config.ts [hunk 4]
```diff
@@ -1,8 +1,18 @@
...
   plugins: [react(), loadVersion()],
+  test: {
+    server: {
+      deps: {
+        // this package does not export esm that uses imports with the file extension, this is a workaround
+        // to instruct vitest to bundle the package similar to the app
+        // https://stackoverflow.com/questions/77126321/how-to-get-vitest-working-with-require-of-es-module
+        inline: ['@ugrc/utah-design-system'],
+      },
+    },
```
Comment (steveoh) [inline vite.config.ts:16]: **suggestion**: i just released the packages with the .js extension if you want to try updating and removing this.

File: src/components/ResultsGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        dataSource: {
+          type: 'queryTable',
+          workspaceId: config.dynamicWorkspaceId,
+          query: removeIrrelevantWhiteSpace(query),
+          oidFields: config.fieldNames.EVENT_ID,
+        },
+      },
+    }),
+    outFields: '*',
+    where: '1 = 1',
```
Comment (steveoh) [inline src/components/ResultsGrid.tsx:72]: this will remove some unnecessary %20 space encoding.
```suggestion
    where: '1=1',
```

**Review by steveoh — COMMENTED** [review]
looks pretty normal. Did the preview fail with the latest changes?

File: firebase.json [hunk 6]
```diff
@@ -66,14 +66,22 @@
...
         ]
       }
+    ],
+    "rewrites": [
+      {
+        "source": "**",
+        "destination": "/index.html"
+      },
+      {
+        "source": "/maps",
```
Comment (steveoh) [inline firebase.json:76]: this may need to be a bit more generic

```suggestion
        "source": "/maps/**",
```

File: package.json (hunk 1 above)
Comment (stdavis) [reply package.json:21]: run-p seemed like it had issues shutting down the firebase emulators. I always had to press control-c twice. Then I looked it up on npm and it hasn't had a new release in something like six year so I tried something more current (pun intended).

File: src/components/utils.ts (hunk 3 above)
Comment (stdavis) [reply src/components/utils.ts:1]: Yes, I just copied what you had which I believe referenced a function in utils that used this import.

File: vite.config.ts (hunk 4 above)
Comment (stdavis) [reply vite.config.ts:16]: The new versions appears to have fixed this issue.

**Comment by stdavis** [comment]
@steveoh It appears that passing the alias for the project id works as expected. This is ready for another look when you have a minute.

> What's a good filter to use to view the table?

Selecting "Disease certification" and "Presence/absence" should bring up a bunch of data in the table.

File: functions/index.js [hunk 7]
```diff
@@ -7,18 +7,19 @@ const options = {
   verbose: !!process.env.FIREBASE_FUNCTIONS_EMULATOR,
   mappings: [
     {
-      from: /^\/toolbox/,
-      to: '/arcgis/rest/services/Electrofishing/Download/GPServer',
+      // the optional /maps prefix is to support the maps rewrite rule in firebase.json
```
Comment (steveoh) [inline functions/index.js:10]: **question**: how is it optional?

**Review by steveoh — COMMENTED** [review]
looks good. At what point do we want to bring the table code back into the design system?

**Comment by stdavis** [comment]
> At what point do we want to bring the table code back into the design system?

I was just waiting for a release with the table component in it.

File: functions/index.js (hunk 7 above)
Comment (stdavis) [reply functions/index.js:10]: Because of the `?` immediately following the capture group.

**Comment by steveoh** [comment]
> I was just waiting for a release with the table component in it.

Sounds good.

**Pull request merged**